ssz-rs/src/array.rs

```diff
@@ -80,6 +80,7 @@ macro_rules! define_ssz_for_array_of_size {
                     for (i, elem) in self.iter_mut().enumerate() {
                         let chunk = elem.hash_tree_root()?;
                         let range = i * BYTES_PER_CHUNK..(i + 1) * BYTES_PER_CHUNK;
+                        // index is safe as long as Node::len() == BYTES_PER_CHUNK
```

Comment:
I think this assumption is strong enough to not warrant a comment but TBH this helps anyone who doesn't know the SSZ spec that well so I don't mind if we leave it in

Reply:
I'm happy to remove it with `Node`'s length being defined as `BYTES_PER_CHUNK`